docs/ja/feature_tap_dance.md

```diff
@@ -0,0 +1,544 @@
+
+次に、いくつかのタップダンスのキーを定義するためには、`TD()` マクロ —— `tap_dance-actions` 配列のインデックスを引数にした `F()` マクロに似ています —— を使うのが最も簡単です。 
+
+その後、`tap_dance_actions` 配列を使って、タップダンスキーを押した時のアクションを定義します。
+
+* `ACTION_TAP_DANCE_DOUBLE(kc1, kc2)`: 1回タップすると `kc1` キーを送信し、そうでなければ `kc2` キーを送信します。キーを押し続けているときは、`kc1` キーを押し続けていることになり、一度タップしてから続けてもう一度キーを押してそのまま押し続けたときは、 `kc2` キーを押し続けていることになります。
+* `ACTION_TAP_DANCE_LAYER_MOVE(kc, layer)`: 1回タップすると `kc` キーコードが送信され、2回タップすると `layer` レイヤーに移動します. (この関数は `TO` のようなものです).
+    * この機能は `ACTION_TAP_DANCE_DUAL_ROLE` と同じですが、機能が明確になるように関数名を変更しました。どちらの関数名でも実行できます。
+* `ACTION_TAP_DANCE_LAYER_TOGGLE(kc, layer)`: 1回タップすると `kc` キーコードが送信され、2回タップすると `layer` レイヤーにトグルします. (この関数は `TG` のようなものです)
+* `ACTION_TAP_DANCE_FN(fn)`: キーマップに定義したキーを、タップダンス実行の回数分タップすると、最後の時点で特定の関数が呼び出されます。
+* `ACTION_TAP_DANCE_FN_ADVANCED(on_each_tap_fn, on_dance_finished_fn, on_dance_reset_fn)`: キーマップに定義したキーをタップする度に最初の関数が呼び出され、タップダンスの実行が終わった時点で2番目の関数が呼び出され、 プ段数の実行をリセットするときに最後の関数が呼び出されます。
+* `ACTION_TAP_DANCE_FN_ADVANCED_TIME(on_each_tap_fn, on_dance_finished_fn, on_dance_reset_fn, tap_specific_tapping_term)`: この関数は `ACTION_TAP_DANCE_FN_ADVANCED` と同じです。しかし、`TAPPING_TERM` で事前に定義した時間に代えて、カスタマイズした時間を使えます。
+
+最初のオプションで、1つのキーに2つの役割を持たせる大抵のケースには十分です。例えば、`ACTION_TAP_DANCE_DOUBLE(KC_SPC, KC_ENT)` は、1回タップすると `Space` を送信し、それ以外の場合は `Enter` を送信します。
+
+!> ここでは [basic keycodes](keycodes_basic.md) のキーだけがサポートされていることを覚えておいてください。カスタムキーコードはサポートされていません。
+
+最初のオプションに似ている2番目のオプションは、単純なレイヤースイッチングのケースに適しています。
+
+これ以上に複雑なケースの場合、3番目か4番目のオプションを使います。（以下でそれらの例を列挙します）
+
+最後に、5番目のオプションは、もし、タップダンスキーをコードに追加した後、非タップダンスキーが奇妙な振る舞いを始めた時に特に役に立ちます。ありうる問題は、あなたがタップダンスキーを使いやすくするために `TAPPING_TERM` の時間を変更し、そのことがその他のキーが割り込みを処理する方法を変更するというものです。
+
+
+## 実装の詳細
+
+さて、説明の大部分はここまでです！　以下に挙げているいくつかの例に取り組むべきです。そして、あなた自身のタップダンスの機能を開発しましょう。しかし、もし、あなたが裏側で起きていることをより深く理解したいのであれば、続けて全ての作品の説明を読みましょう！
+
+メインエントリーポイントは、`process_tap_dance()` で、キーを押すたびに `process_record_quantum()` から呼び出され、ハンドラが素早く実行されます。この関数は、押されたキーがタップダンスキーがどうか確認します。
+もし、押されたキーがタップダンスキーではなく、かつ、タップダンスが実行されていたなら、最初に、新しく押されたキーをキューに格納します。
+もし、押されたキーがタップダンスキーであるなら、それから既に押されていたキーと同じキーか確認します（if there's one active, that is）。
+もし、既に押されたキーと同じでない場合、まず、既に押されたキーを送信し、続いて新しいキーを登録します。
+もし、既に押されたキーと同じであった場合、カウンタの値を増やし、タイマーをリセットします。
+
+このことは、あなたは再びキーをタップするまでの時間として `TAPPING_TERM` の時間を持っていることを意味します。そのため、あなたは1つの `TAPPING_TERM` の時間内に全てのタップを行う必要はありません。このことは、より長いタップ回数を許可すると、キーの反応に最小限の影響があるということです。 
+
+次の停車駅は `matrix_scan_tap_dance()` です。この関数はタップダンスキーの時間切れを制御します。
+
+柔軟性のために、タップダンスは、キーコードの組み合わせにも、ユーザー関数にもなることができます。後者は、より高度はタップ回数の制御や、LED を点滅させたり、バックライトをいじったり、等々の制御を許可します。これは、1つの調和と、いくつかの賢いマクロによって成し遂げられています。
+
+# 実装例
+
+## シンプルな実装例
+
+ここに1つの定義のための簡単な例があります。
+
+1. `rules.mk` に `TAP_DANCE_ENABLE = yes` を追加します。
+1. `config.h` ファイル（キーマップディレクトリにある `qmk_firmware/keyboards/planck/config.h` からコピーできます）に `#define TAPPING_TERM 200` を追加します。
```

Comment:
```suggestion
2. `config.h` ファイル（キーマップディレクトリにある `qmk_firmware/keyboards/planck/config.h` からコピーできます）に `#define TAPPING_TERM 200` を追加します。
```

Reply:
Markdown is either smart enough to understand what you want here, or ignores the actual numbers completely. But/so, I'm not sure if we should rely on that.
![image](https://user-images.githubusercontent.com/4781841/74202855-ca9e6a00-4cc1-11ea-96c0-de0f907c2e41.png)

Reply:
Oh interesting, Markdown knowledge deepened.

Reply:
As explained by fauxpark, there is no problem with the current code, so we will not change it.

![image](https://user-images.githubusercontent.com/20755159/74213498-da777780-4cdc-11ea-837a-4d18c28aef84.png)

Reply:
Works for me.

Reply:
I would prefer it was changed, actually. What I meant is that we can't rely on this behaviour being the case, so better to number them manually.

Reply:
OK. Assign a sequence number manually, as it is not guaranteed that a sequence number will always be assigned.